buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlTestPlugin.java

```diff
@@ -0,0 +1,126 @@
+
+        // create task - note can not use .register due to the work in RestIntegTestTask's constructor :(
+        RestIntegTestTask yamlTestTask = project.getTasks()
+            .create(
+                SOURCE_SET_NAME,
+                RestIntegTestTask.class,
+                task -> { task.dependsOn(project.getTasks().getByName("copyRestApiSpecsTask")); }
+            );
+        yamlTestTask.setGroup(JavaBasePlugin.VERIFICATION_GROUP);
+        yamlTestTask.setDescription("Runs the YAML based REST tests against an external cluster");
+
+        // setup task dependency
+        project.getDependencies().add(SOURCE_SET_NAME + "Compile", project.project(":test:framework"));
+
+        // setup the runner
+        RestTestRunnerTask runner = (RestTestRunnerTask) project.getTasks().getByName(yamlTestTask.getName() + "Runner");
+        runner.setTestClassesDirs(yamlTestSourceSet.getOutput().getClassesDirs());
+        runner.setClasspath(yamlTestSourceSet.getRuntimeClasspath());
+
+        // if this a module or plugin, it may have an associated zip file with it's contents, add that to the test cluster
+        boolean isModule = project.getPath().startsWith(":modules:");
+        Zip bundle = (Zip) project.getTasks().findByName("bundlePlugin");
+        if (bundle != null) {
+            yamlTestTask.dependsOn(bundle);
+            if (isModule) {
+                runner.getClusters().forEach(c -> c.module(bundle.getArchiveFile()));
+            } else {
+                runner.getClusters().forEach(c -> c.plugin(project.getObjects().fileProperty().value(bundle.getArchiveFile())));
+            }
+        }
+
+        // es-plugins may declare dependencies on additional modules, add those to the test cluster too.
+        project.afterEvaluate(p -> {
+            PluginPropertiesExtension pluginPropertiesExtension = project.getExtensions().findByType(PluginPropertiesExtension.class);
+            if (pluginPropertiesExtension != null) { // not all projects are defined as plugins
+                pluginPropertiesExtension.getExtendedPlugins().forEach(pluginName -> {
+                    Project extensionProject = project.getProject().findProject(":modules:" + pluginName);
+                    if (extensionProject != null) { // extension plugin may be defined, but not required to be a module
+                        Zip extensionBundle = (Zip) extensionProject.getTasks().getByName("bundlePlugin");
+                        yamlTestTask.dependsOn(extensionBundle);
+                        runner.getClusters().forEach(c -> c.module(extensionBundle.getArchiveFile()));
+                    }
+                });
+            }
+        });
+
+        // setup IDE
+        GradleUtils.setupIdeForTestSourceSet(project, yamlTestSourceSet);
+
+        // run test tasks first if they exist since they are presumably faster and less resources intensive
+        Task testTask = project.getTasks().findByName("test");
```

Comment:
Let's remove this. I don't think it should go here, plus using `findByName` is going to eagerly realize the task.

Reply:
updated it to be inline with https://github.com/elastic/elasticsearch/pull/57016